awx/main/models/events.py

```diff
@@ -546,6 +548,13 @@ def _update_host_summary_from_stats(self, hostnames):
 
             Host.objects.bulk_update(list(updated_hosts), ['last_job_id', 'last_job_host_summary_id'], batch_size=100)
 
+            # bulk-create
```

Comment:
This looks about right to me.

A good place to add some tests for logic would be:

`awx/main/tests/functional/models/test_events.py`

There's similar tests in here that validate the creation of the `JobHostSummary` objects.

Reply:
I have made the required changes. I have added a unit test case in the file : 
```awx/main/tests/functional/models/test_host_metric.py```